pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -76,10 +76,15 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
 
     /**
      * @brief Randomly set the seed of the internal random generator
+     * If catalyst already provides a random generator, use the provided one
      *
-     * @param seed Seed
+     * @param catalyst_rng Seed
      */
-    void setRandomSeed() {
+    void setRandomSeed(std::mt19937 *catalyst_rng = nullptr) {
+        if (catalyst_rng){
+            this->rng = *catalyst_rng;
```

Comment:
Is `this->` necessary?

Reply:
You are right, I think we don't need `this->`

Reply:
Actually I would like to keep `this->` just to be explicit, if that's fine : ) 

(Also the variable name `catalyst_rng` has changed to `rng`)

Reply:
This function no longer exists